projects/m2/003-remove-outliers/solutions/andreaattardo/solution_remove-outliers.c

```diff
@@ -0,0 +1,155 @@
+//remove-outliers.c
+// ---------------
+//When analysing data collected as part of a science experiment it may be desirable to remove the most extreme values before performing other calculations.
+//Write a function that takes a list of values and an non - negative integer, n, as its parameters.The function should create a new copy of the list with the n largest
+//elementsand the n smallest elements removed.Then it should return the new copy of the list as the function�s only result.
+
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+//CONSTANTS
+//----------
+#define USRINPUT_LEN 50 //usr input stream length
+
+
+//FUNCTIONS DECLARATIONS
+// --------------------
+int* sortIntArray(int* intArray, int arrayElements);
+int getInt();
+int* initDynamicArray(int arrayElements);
+void printIntArray(int* intArray, int arrayElements);
+int* removeOutliers(int* intArray, int arrayElements, int nOutliers);
+
```

Comment:
qui spezzerei l'esercizio in 2 file: la parte da //CONSTANTS fino a questa riga la metterei in un file chiamato solution_remove-outliers.h e poi nel presente file metterei solo #include "filediprima.h" subito dopo l'inclusione dei file standard di libreria